hudi-hive/src/main/java/org/apache/hudi/hive/HoodieHiveClient.java

```diff
@@ -192,7 +192,7 @@ private String getPartitionClause(String partition) {
     String alterTable = "ALTER TABLE " + syncConfig.tableName;
     for (String partition : partitions) {
       String partitionClause = getPartitionClause(partition);
-      String fullPartitionPath = FSUtils.getPartitionPath(syncConfig.basePath, partition).toString();
+      String fullPartitionPath = FSUtils.getURIPath(fs, FSUtils.getPartitionPath(syncConfig.basePath, partition));
```

Comment:
I would add a check here to check `fs.uri` and do this only if it is `hdfs` and not in case of other file systems. It also makes it explicitly clear why we are doing this.